### DISCUSSION TITLE
Black screen - after re-flashing SNES repro

### DISCUSSION THREAD

**Discussion by StanMOI** [sanni/cartreader#148] [Need help]
Hello Sanni,

i trying to reflash Chinese Repro PCB.
Flash Eprom ; MX29LV640 
Read PCB = works 
Write PCB = works but black screen on my tv.

i tried to use same rom that was inside before (and witch worked just before erase),
Also tried to use other rom’s = same problem.
Same problem with HiRom and LoRom PCB.
I suspect « region unlock chip ».
What do you think ?

**Comment by sanni** [comment]
Try removing the soldered cable to wr again and add the solder bridge like before. Maybe this repro needs it somehow.

**Reply by StanMOI** [reply]
Already done.
Nothing changes :(

is there some possibility with your programmer to do test for « region unlock chip » (12f629) ?

PS: 

- I desoldered SUPER CIC 12f629 and put on PCB one from an other worked PCB = still have same problem of black screen.
- I tried to Write same game with NTSC and PAL region = all same, black screen.

Dont know what to do anymore. 
:(

**Reply by JadersonAragao** [reply]
StanMOI como vc conseguiu dividir a rom em 2 arquivos ?

**Comment by StanMOI** [comment]
Hi Sanni,
I come back to you again Sanni with this diagram of SNES Repros that I made.
I think there is some problem with pinouts that make something wrong.
Maybe some wrong connections are making "bad erase" of data when I trie to "erase" chip.
Please, can you check for me and see if you can find some wrong connections that can do problem with "not full erasing" or "not full writing".

<img width="1142" alt="@stanmoi25 SNES Repro Flash EPROM pinouts" src="https://user-images.githubusercontent.com/38383423/117810950-4b9b3700-b260-11eb-86c4-dabf98eced7d.png">


If I compare with your TSOP48 adapter pinouts, I can see that on my PCB those pins are not connected or with some different connections:
**REPRO:**
CART 18 "/IRQ" = > not connected
**On your adapter:**
CART 18 "/IRQ" = > FLASH pin11 "WE#" 

**REPRO:**
FLASH pin13 = > CART 48 
**On your adapter :**
FLASH pin13 = > CART 47

**HiROM REPRO:**
CART 47 "BA6" = > Connected to SRAM (64Kbit, HiROM) pin13
**LoROM REPRO:**
CART 47 "BA6" = > not Connected
**On your adapter :**
CART 47 "BA6" = > FLASH pin13

<img width="857" alt="Capture d’écran 2021-05-11 à 02 22 46" src="https://user-images.githubusercontent.com/38383423/117742463-26c5a600-b205-11eb-9b65-ad34977d2a1c.png">

I tried to connect on my HiROM REPRO "CART 48" to "CART 47" then to re-erase and to re-program it again, still have the black screen.
Have you any idea how to resolve it ?
Please let me know.

**Comment by sanni** [comment]
What game was originally on the repro? And what game do you want to flash?

The IRQ pin doesn't matter, the A22/A23 ist because your repro seems to share the same PCB as ExHiROM/6MB games which is currently not supported by the flashrom function. Currently you can only write SNES games up to 4MB. 
You can try limiting the flash size to 4MB.

Change:
````
println_Msg(F("MX29LV640 detected"));
println_Msg(F("ATTENTION 3.3V"));
flashSize = 8388608;
````
to:
````
println_Msg(F("MX29LV640 detected"));
println_Msg(F("ATTENTION 3.3V"));
flashSize = 4194304;
````

See if that changes anything.

Also where is the OE# (FLASH) pin connected to on the HiROM PCB?

**Reply by StanMOI** [reply]
> What game was originally on the repro? And what game do you want to flash?

"Tales of Phantasia" ExHiROM/6MB
Now I see better why originally file couldn't be re-writed (cause it's not supported).
But I also tried to flash just HiROM game with same SRAM (64Kbit), with no success.

> Also where is the OE# (FLASH) pin connected to on the HiROM PCB?

on the HiROM PCB =>  CART 49 "/CART"
on the LoROM PCB =>  74HC139 pin7  "1Y3"

PS: I corrected it on my diagram picture.


> Change to **flashSize = 4194304;**

I done it and make some tests = always same problem.

Just to let you know.
I made many tests with something about 20 ROM's.
I found one ROM worked without any changes "Stone Protectors (U)" : 16Mbit LoROM-SRAM 256
This ROM work's without any changes like : flashSize = 4194304 (vs 8388608), or without adding connection between "CART 48" and "CART 47".

**Comment by sanni** [comment]
You can try the following:
- Update the Cart Reader to the [new version](https://github.com/sanni/cartreader/archive/refs/heads/master.zip) released today.   
- You now find two HiROM options P0 and P1 in the new repro menu.    
- Split Tales of Phantasia into two files, the first 4MB of the ROM into one file and the second 2MB into another file.   
- Select P0, erase the chip and flash the 2MB file.    
- Switch Cart Reader off/on, select P1, do **not** erase again, and flash the 4MB file.    

Not sure if it will work though since I can't test it.

**Reply by StanMOI** [reply]
> You can try the following:
> 
> * Update the Cart Reader to the [new version](https://github.com/sanni/cartreader/archive/refs/heads/master.zip) released today.
> * You now find two HiROM options P0 and P1 in the new repro menu.
> * Split Tales of Phantasia into two files, the first 4MB of the ROM into one file and the second 2MB into another file.
> * Select P0, erase the chip and flash the 2MB file.
> * Switch Cart Reader off/on, select P1, do **not** erase again, and flash the 4MB file.
> 
> Not sure if it will work though since I can't test it.

Hi Sanni,
Method to split rom file "Tales of Phantasia" (6MB) into two files 4MB and 2MB = **WORK’S** !
I tried to write Daikaijuu 2 (6MB) with same method and it's work's fine!
Now I can confirm that I can rewrite my ex-HiROM pcb cartridge with ex-HiROM files.

But I still can not write on this ex-HiROM board an simple HiROM game.
This must be caused by PCB design witch are maybe only for ex-HiROM games.
Maybe need some to unsolder some jumpers ...

**LoROM after update v6.1:**
Unfortunately, after this update, now when I put LoROM PCB it's says "unknown Flashrom" :(
So I can not test it.

**Comment by JadersonAragao** [comment]
Como dividir a rom em 2 arquivos?

**Reply by StanMOI** [reply]
Use English translation

**Reply by JadersonAragao** [reply]
How to split rom into 2 files?

**Reply by StanMOI** [reply]
Here you can find softwares for SNES rom's.
"SFC/SNES ROM Utility" have this option.
https://drive.google.com/drive/folders/1atC0UN0l9HXI9xFH8b58W32HNxo9x-p8?usp=sharing
![593screenshot1](https://user-images.githubusercontent.com/38383423/168819672-ce594c9f-e7d5-4802-ae81-b799b26573b2.png)

Or use any windows/Mac software to split file.
![splitter](https://user-images.githubusercontent.com/38383423/168820513-4dd0c2a2-f9cb-4807-9f6c-d0de3f41e4a5.jpg)

**Comment by JadersonAragao** [comment]
thank you so much. StanMOI
